### PR TITLE
Make possible to enable unit file via symlink (v4)

### PR DESCRIPTION
**Pull request by msekletar**
Changes since v3:
- only bug fixes reported for #910

### CODE REVIEW

File: src/basic/path-util.c [hunk 1]
```diff
@@ -853,3 +854,28 @@ char *prefix_root(const char *root, const char *path) {
...
+        while ((p = va_arg(ap, const char*)))
+                if (path_startswith(path, p)) {
+                        va_end(ap);
+                        return true;
+                }
+
+        va_end(ap);
+
+        return false;
+}
```
Comment (zonque) [inline src/basic/path-util.c:881]: This seems overly complicated, no? Why treat `x` special, and not just walk the list? If it's empty, the loop will be a noop. Also, you can have a local bool variable called `found` or something, and if `path_startswith()` returns success, just latch it and break from the loop. That would make the code contain just one `va_end()`. Makes sense, or do I miss anything?

Comment (zonque) [inline src/basic/path-util.c:881]: That is, the function would omit `x` all toghether:

``` c
bool path_prefix_in_list(const char *path, ...) {
```

Comment (msekletar) [inline src/basic/path-util.c:881]: Idea was that if you want to call with empty list then you have to explicitly specify NULL as second argument. I think we do the same for some of our public APIs, for example,

``` C
int sd_bus_emit_interfaces_added(sd_bus *bus, const char *path, const char *interface, ...) _sd_sentinel_
```

I figured that doing the same would be probably OK.

Comment (zonque) [inline src/basic/path-util.c:881]: Sure, that should work just fine. APIs should have all mandatory parameters listed as variables, but all optional ones as `...`. The only exception is with functions that only have optional ones. In this case, a dummy variable has to be added, such as in `strjoin()`.

If you call `va_start(ap, path);`, and the first optional parameter after `path` is `NULL`, `va_arg()` will return `NULL` in its first invocation.

**Comment by poettering** [comment]
OK, @msekletar just let me know that a new version is already in the works, won't finish the review then.

**Comment by msekletar** [comment]
Closed in favor of #1240

**Pull request closed without merging**